ppa/archive/management/commands/generate_textcorpus.py

```diff
@@ -0,0 +1,200 @@
+"""
+**generate_textcorpus** is a custom manage command to generate a plain
+text corpus from Solr.  It should be run *after* content has been indexed
+into Solr via the **index** manage command.
+"""
+
+import os
+import json
+from django.core.management.base import BaseCommand
+from parasolr.django import SolrQuerySet
+from collections import defaultdict, OrderedDict
+import logging
+from tqdm import tqdm
+from typing import Tuple
+from contextlib import contextmanager
+import logging
+
+class SolrCorpus:
```

Comment:
I vaguely remember this class from Vineet's code. It's great to keep it if you find it a useful way to encapsulate the logic. (I would probably put the methods on the manage command, personally.)

If you keep it, you should write unit tests for each method of the class.